pkg/leaderelection/leaderelection.go

```diff
@@ -144,7 +144,9 @@ type LeaderElector struct {
 // Run starts the leader election loop
 func (le *LeaderElector) Run(stop <-chan struct{}) error {
 	defer func() {
-		le.config.Callbacks.OnStoppedLeading()
+		if err := le.config.Callbacks.OnStoppedLeading(); err != nil {
+			le.logger.Sugar().Errorf("unable to stop properly leading, error:%v", err)
```

Comment:
Not sure if it is useful but you can modify the main func returned error by closure if you name it.

Reply:
Yes thanks, I will do it